File: leetcode/array/53.maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        for i in range(len(nums)):
            if i == 0:
                continue
            elif nums[i] + nums[i-1] < nums[i]:
                continue
            else:
                nums[i] = nums[i-1] + nums[i]
        return max(nums)

File: leetcodeJS/array/53.maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function (nums) {
  if (nums.length === 0) return 0;
  let solution = Number.NEGATIVE_INFINITY;
  let temp = Number.NEGATIVE_INFINITY;
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] > temp + nums[i]) temp = nums[i];
    else {
      temp += nums[i];
    }
    if (temp > solution) solution = temp;
  }

  return solution;
};
